lib/BuildSystem/BuildSystem.cpp

```diff
@@ -1203,6 +1205,13 @@ BuildSystemImpl::lookupNode(StringRef name, bool isImplicit) {
 }
 
 llvm::Optional<BuildValue> BuildSystemImpl::build(BuildKey key) {
+
+  // If we were cancelled, return.
+  if (isCancelled_) {
+      isCancelled_ = false;
+      return None;
+  }
+
   // Create the execution queue.
   executionQueue = delegate.createExecutionQueue();
 
```

Comment:
May be we can combine these and have a status enum here. `{Started, QueueCreated, Aborted, Cancelled}` but not sure what are the exact semantics of BuildSystem.